src/device.ts

```diff
@@ -312,15 +312,24 @@ export class Device extends vscode.Disposable {
      */
     public async mkdir_p(path: string): Promise<void> {
         const names = path.split('/');
+
+        // If path had a preceding slash, remove the empty path segment
+        if(!names[0]) {
+            names.shift();
+        }
+
         let part = '';
         while (names.length) {
-            part += names.shift() + '/';
-            // have to make sure the directory exists on the remote device first
+            part += '/' + names.shift();
+            // Create the directory if it doesn't already exist
             try {
-                await this.stat(part);
+                const stat = await this.stat(part);
+                if (!stat.isDirectory()) {
+                    throw new Error("Cannot create directory: part of path exists but isn't a directory");
```

Comment:
It could be helpful to include the path in the error message, for example:

    `Cannot create '${path}': '${part}' exists but is not a directory`

Reply:
I thought you said you wanted to keep error messages short :laughing:

Reply:
I know. I was thinking about that when I wrote it. :laughing: 

I'll leave it up to your best judgment here. Whatever you think would help the user the most.

Reply:
Hmmm... I wonder if there's a way to do multiline messages?

Reply:
I split the difference: now it specified the part that it failed to create but doesn't include the whole path that was going to be created.